### ISSUE TITLE
Having problems with user authentication 

### ISSUE THREAD

**Issue by MohakAhuja21** [meabhisingh/mernProjectEcommerce#44]
I am stuck at user authentication part. If someone can help me out I'll be more than helpful🙏

**Comment by codeomi** [comment]
Whats the issue?

**Comment by Souvik-Halder** [comment]
can you just briefly describe about the user authentication issue . is this issue related to the load user?

**Comment by MohakAhuja21** [comment]
> can you just briefly describe about the user authentication issue . is this issue related to the load user?

yes. i cannot load user. everytime i refresh. it just takes away user and it gets to undefined.

**Comment by Souvik-Halder** [comment]
Okay, so in that case the issue is with the cookie itself you need to get the cookie in the fronend itself 
to do that follow the below code in frontend folder open App.js
Write the below code there
`import axios from "axios";`
`axios.defaults.withCredentials=true`

**Comment by Souvik-Halder** [comment]
If you are using the cors in the backend just let me know that. If your backend configuration is not done properly then it will throw an error after modification in the frontend folder by the upper code. So just let me know about the cors configuration in the backend

**Comment by MohakAhuja21** [comment]
Where can I contact you ?

**Comment by Souvik-Halder** [comment]
send me your repo

**Comment by Souvik-Halder** [comment]
@meabhisingh please assign me the issue I can handle this

**Comment by MohakAhuja21** [comment]
Why don't you make a pull request to the repository ??

**Comment by Souvik-Halder** [comment]
Okay I will do that

**Comment by skjilani9** [comment]
> Okay I will do that

bro iam not getting cookies in chrome any solution for it

**Comment by ghost** [comment]
bro i am not getting cookies in chrome any solution for it

**Comment by Souvik-Halder** [comment]
import axios from "axios";
axios.defaults.withCredentials=true
use this code in app.js

**Comment by ghost** [comment]
Bro I am not getting cookies in backend

**Comment by Priyansh6570** [comment]
> import axios from "axios"; axios.defaults.withCredentials=true use this code in app.js

i had same issue that upon refresh i was loged out, so i tried to implement your solution, but as i am using cors in my backend i am having issue with origin after setting credentials=true in App.js in frontend.
following is code from my app.js from backend where i used cors

```
import cors from 'cors';
app.use(cors());
```
**EDIT :**  i solved my cors error i just had to manage my header request to allow them from frontend, for that i made following adjustments to my cors

`app.use(cors({
    origin: 'http://localhost:5173', // replace with your frontend URL
    credentials: true,
    allowedHeaders: ['Content-Type', 'Authorization'], // add headers that you want to allow
    methods: ['GET', 'POST', 'PUT', 'DELETE'], // add HTTP methods that you want to allow
    optionsSuccessStatus: 200 // return 200 status code for preflight requests
  })); `
  This worked for me !
  and Thenks @Souvik-Halder for providing frontend App.js changes, it really helped

**Comment by mdsujan-mridha** [comment]
If anyone face this type issue 
then open Front-end>open app.js
then past below code :
import axios from 'axios';
axios.defaults.withCredentials = true;
after that open Backend>open app.js
then past below code:
const cors = require('cors');
const corsOptions ={
    origin:'http://localhost:3000', 
    'Content-Type': 'Authorization',
    credentials:true,          
    optionSuccessStatus:200
}
app.use(cors(corsOptions));

**Comment by pushpendarchoudhary** [comment]
after user login as soon as i am switching between component authentication become false

**Comment by Anmol-002** [comment]
@mdsujan-mridha  @Souvik-Halder bro can anyone of you help me out. I am facing the same issue. I have uploaded my backend on render , if i am using postman i am seeing my token but when i using my hosted backend whenever i login and go to account section, on refresh i am logged out. Because of this even after login i can see my orderes or dashboard its saying log in to access this resource. [https://github.com/Anmol-002/E-commerce](This) is my repo or u can see in my profile by  name of ecommerce
